opentracing/src/main/java/opentracing/propagation/Extractor.java

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright 2016 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package opentracing.propagation;
+
+import opentracing.Span;
+
+/**
+ */
+public interface Extractor<T> {
```

Comment:
Per my comment (https://github.com/opentracing/opentracing-java/pull/11#discussion_r54204792) on #11: without taking a particular stance on any of the "internal" APIs, it would simplify (and accelerate) this review process if we could focus on the public-facing aspects of *this* PR and get it in as a much-needed stake-in-the-ground for `opentracing-java`. Sound ok?